CONTRIBUTING.md

```diff
@@ -110,8 +110,8 @@ codebase follows these rules, things are work in progress.
     our code may contain additions which are not compatible with the STL (e.g. `iox::cxx::vector::emplace_back()`
     does return a bool)
 7) **Always use `iox::log::Logger`**, instead of `printf()`
-8) **Always use `iox::ErrorHandler()`**, when an error occurs that cannot or shall not be propagated via an
-    `iox::cxx::expected`, the `iox::ErrorHandler()` shall be used; exceptions are not allowed
+8) **Always use `iox::ErrorHandler()` or `cxx::Expects`/`cxx::Ensures`**, when an error occurs that cannot or shall not be propagated via an
```

Comment:
In my opinion this will be very soon outdated when the low level error handler becomes also available for iceoryx hoofs. Then it would make sense to never use `cxx::Expects` / `cxx::Ensures` (even when they use the error handler behind the scenes).
I would state here right from the beginning: always use the `iox::ErrorHandler` since we can emit an error code here which can be important for later bug reports or debugging.

Reply:
I don't think it makes sense to use `errorHandler()` in `iceoryx_hoofs` building blocks. IMHO it's more idiomatic and closer to a library philosophy to use `Expects` / `Ensures` and allow the users to call the `errorHandler()` inside `Expects` / `Ensures` (or `throw` or `std::terminate`).

However, we should not discuss this here, that's something for #1032

The goal for `v2.0` is to decouple `iceoryx_posh` and `iceoryx_binding_c` from `iceoryx_hoofs` as an intermediate step. Hence, the description above is still valid.

cc @MatthiasKillat